Utils/ScriptValidator/Unit1.pas

```diff
@@ -30,21 +34,37 @@   TForm1 = class(TForm)
 
 var
   Form1: TForm1;
-
-
+  IsValidatePath : Boolean;
+  sListFileInFolder : TStringList;
 implementation
 uses
   KM_Maps;
 
 {$R *.dfm}
 
+procedure TForm1.FindFiles(Path: String; out aList: TStringList);
+var SearchRec:TSearchRec;
+begin
+  FindFirst(Path+PathDelim+'*', faAnyFile, SearchRec);
+  repeat
+    if (SearchRec.Name <> '.') and (SearchRec.Name <> '..') then
+      if (SearchRec.Attr and faDirectory = faDirectory) then
+        FindFiles(Path + PathDelim + SearchRec.Name, aList)
+      else
+        if SameText(RightStr(SearchRec.Name, Length(EXT_FILE_SCRIPT)+1), ('.'+EXT_FILE_SCRIPT)) then

```

Comment:
Looks quite complicated.
Should be SameText(ExtractFileExt(SearchRec.Name), EXT_FILE_SCRIPT)